optuna/study.py

```diff
@@ -580,51 +581,67 @@ def objective(trial):
                 Parameter values to pass your objective function.
         """
 
-        system_attrs = {"fixed_params": params}
-        self._append_trial(state=TrialState.WAITING, system_attrs=system_attrs)
-
-    def _append_trial(
-        self,
-        value=None,  # type: Optional[float]
-        params=None,  # type: Optional[Dict[str, Any]]
-        distributions=None,  # type: Optional[Dict[str, BaseDistribution]]
-        user_attrs=None,  # type: Optional[Dict[str, Any]]
-        system_attrs=None,  # type: Optional[Dict[str, Any]]
-        intermediate_values=None,  # type: Optional[Dict[int, float]]
-        state=TrialState.COMPLETE,  # type: TrialState
-        datetime_start=None,  # type: Optional[datetime.datetime]
-        datetime_complete=None,  # type: Optional[datetime.datetime]
-    ):
-        # type: (...) -> int
-
-        params = params or {}
-        distributions = distributions or {}
-        user_attrs = user_attrs or {}
-        system_attrs = system_attrs or {}
-        intermediate_values = intermediate_values or {}
-        datetime_start = datetime_start or datetime.datetime.now()
-
-        if state.is_finished():
-            datetime_complete = datetime_complete or datetime.datetime.now()
-
-        trial = FrozenTrial(
-            number=-1,  # dummy value.
-            trial_id=-1,  # dummy value.
-            state=state,
-            value=value,
-            datetime_start=datetime_start,
-            datetime_complete=datetime_complete,
-            params=params,
-            distributions=distributions,
-            user_attrs=user_attrs,
-            system_attrs=system_attrs,
-            intermediate_values=intermediate_values,
+        self.add_trial(
+            create_trial(state=TrialState.WAITING, system_attrs={"fixed_params": params})
         )
 
+    @experimental("2.0.0")
+    def add_trial(self, trial: FrozenTrial) -> None:
+        """Add trial to study.
+
+        The trial is validated before being added.
```

Comment:
I see your point. On the other hand, it should be clear from the type hint and I think this way of putting it is consistent across Optuna. Please correct me otherwise.

Reply:
Thanks for the explanation. I've understood it. We don't need to note it. Thanks!